pages/login.vue

```diff
@@ -1,3 +1,83 @@
+<script setup>
+    // import { generateCodeVerifier,  generateChallenge } from 'pkce-challenge'
+    import pkceChallenge from 'pkce-challenge'
+
+    const client_id = "20aa48c2719e42c0be5f3b834942f06d"
+    const scopes = []  // https://developer.spotify.com/documentation/general/guides/authorization/scopes/
+    const redirect_uri = new URL(window.location)
+    redirect_uri.search = ""  // clear search stuff
+    const auth_token = localStorage.getItem('auth-token')
+
+    async function logout(){
+        localStorage.removeItem('auth-token')
+        navigateTo('/')
+    }
+
+    async function login(){
+        const pkce = pkceChallenge()
+
+        localStorage.setItem('code-verifier', pkce.code_verifier)
+
+        const url = new URL('/authorize', 'https://accounts.spotify.com/')
+        url.searchParams.append('client_id', client_id)
+        url.searchParams.append('response_type', 'code')
+        url.searchParams.append('redirect_uri', redirect_uri)
+        url.searchParams.append('scopes', scopes.join(" "))
+        url.searchParams.append('code_challenge_method', 'S256')
+        url.searchParams.append('code_challenge', pkce.code_challenge)
```

Comment:
please add a "state" query parameter as specified in the docs. This param should contain a randomly generated string, also stored in local storage. https://developer.spotify.com/documentation/general/guides/authorization/code-flow/